lib/http-server.js

```diff
@@ -64,6 +65,20 @@ function HttpServer(options) {
     res.emit('next');
   });
 
+  if (options.username || options.password) {
+    before.push(function (req, res) {
+      var credentials = auth(req);
+
+      if (credentials && credentials.name === options.username && credentials.pass === options.password) {
```

Comment:
I think in my head I was thinking 'better not make it secure or people might actually use it to secure things!' but that is obviously unhelpful because a) most people won't know what a timing attack is and so won't benefit from knowing it's not secure, and b) well, it just is.

So: great idea, good reasoning. I'll implement it this week.